Contagem da quantidade de crimes violentos ocoridos em Minas Gerais em 2023 (por natureza do crime)

In [8]:
import pandas as pd

df = pd.read_csv('../data/bruto/crimes_violentos_2023.csv', sep= ';')

df.drop_duplicates()

contCrimes = df['natureza'].value_counts()

print(contCrimes)


natureza
Estupro Consumado                        10236
Estupro de Vulnerável Consumado          10236
Estupro de Vulnerável Tentado            10236
Estupro Tentado                          10236
Extorsão Consumado                       10236
Extorsão Mediante Sequestro Consumado    10236
Extorsão Tentado                         10236
Homicídio Tentado                        10236
Roubo Consumado                          10236
Roubo Tentado                            10236
Sequestro e Cárcere Privado Consumado    10236
Sequestro e Cárcere Privado Tentado      10236
Homicídio Consumado (Registros)          10236
Name: count, dtype: int64


Isso acontece porque o arquivo está estruturado com blocos idênticos:
para cada tipo de crime, há uma linha para cada município de Minas Gerais, sempre com o mesmo número de registros (10236).

Crimes por Regiões Integradas de Segurança Pública (RISP)

In [1]:
import pandas as pd

df = pd.read_csv('../data/bruto/crimes_violentos_2023.csv', sep= ';')

df.drop_duplicates()

crimes_regiao = df.groupby('risp')['natureza'].count().sort_values(ascending=False)

crimes_regiao = crimes_regiao.reset_index()
crimes_regiao.columns = ['RISP', 'TOTAL_CRIMES']

crimes_regiao.to_csv('../data/processado/crimes_por_regiao.csv', sep=';', index=False, encoding='utf-8')


Unir Tabela de Crimes com tabela de códigos de regiões

Inserir as coordenadas da cede de cada região (RISP) segundo o Google Maps

In [5]:
import pandas as pd

df = pd.read_csv('../data/processado/crimes_por_regiao.csv', sep=';')

coords = {
    1:  [-19.9167, -43.9345],  # Belo Horizonte
    2:  [-19.7668, -44.0868],  # Ribeirão das Neves
    3:  [-19.6944, -43.9237],  # Vespasiano
    4:  [-21.7642, -43.3496],  # Juiz de Fora
    5:  [-19.7479, -47.9319],  # Uberaba
    6:  [-21.2475, -44.9997],  # Lavras
    7:  [-20.1452, -44.8912],  # Divinópolis
    8:  [-18.8545, -41.9459],  # Governador Valadares
    9:  [-18.9128, -48.2755],  # Uberlândia
    10: [-19.0003, -46.3162],  # Carmo do Paranaíba 
    11: [-16.7286, -43.8578],  # Montes Claros
    12: [-19.4820, -42.5279],  # Ipatinga
    13: [-21.2250, -43.7737],  # Barbacena
    14: [-18.7573, -44.4303],  # Curvelo
    15: [-17.8576, -41.5059],  # Teófilo Otoni
    16: [-16.3595, -46.9067],  # Unaí
    17: [-22.2300, -45.9380],  # Pouso Alegre
    18: [-21.7890, -46.5663],  # Poços de Caldas
    19: [-19.4671, -44.2469],  # Sete Lagoas
}

df['latitude'] = df['RISP'].map(lambda x: coords.get(x, [None, None])[0])
df['longitude'] = df['RISP'].map(lambda x: coords.get(x, [None, None])[1])

df.to_csv('../data/processado/crimes_por_regiao.csv', sep=';', index=False, encoding='utf-8')


Criando o mapa de calor

In [7]:
import folium
from folium.plugins import HeatMap

df = pd.read_csv('../data/processado/crimes_por_regiao.csv', sep=';')

mapa = folium.Map(location= [-18.472096, -44.353396], zoom_start= 7)
coords = list(zip(df.latitude, df.longitude))
mapaCalor = HeatMap(coords, radius=9, blur=10)
mapa.add_child(mapaCalor)

mapa.save('../src/mapa_calor_crimes_MG.html')